lib/src/Uncertainty/Model/RatioOfUniforms.cxx

```diff
@@ -0,0 +1,484 @@
+  infV_.resize(dimension);
+  supV_.resize(dimension);
+  const Point zero(dimension, 0.0);
+  Sample startingPointsInfV(0, dimension);
+  Sample startingPointsSupV(0, dimension);
+  for (UnsignedInteger i = 0; i < dimension; ++ i)
+    {
+      const Function objectiveVI(objectiveV.getMarginal(i));
+      OptimizationProblem problemVI(objectiveVI);
+      problemVI.setMinimization(false);
+      if (ub[i] > 0.0)
+	{
+	  // find a feasible starting point in [0, ub]
+	  if (!startingPointsSupV.getSize())
+	    {
+	      for (UnsignedInteger k = 0; k < candidateNumber_; ++ k)
+		{
+		  Point candidate(sequence.generate());
+		  for (UnsignedInteger j = 0; j < dimension; ++ j)
+		    candidate[j] = candidate[j] * ub[j];
+		  if (SpecFunc::IsNormal(logUnscaledPDF_(candidate)[0]))
+		    {
+		    startingPointsSupV.add(candidate);
+		    if (startingPointsSupV.getSize() == maximumMultistart_)
+		      break;
+		    }
+		} // for k
+	    } // if (!startingPointsSupV.getSize())
+	  if (!startingPointsSupV.getSize())
+	    throw InternalException(HERE) << "Could not find a feasible starting point to initialize ratio of uniforms V sup";
+	  problemVI.setBounds(Interval(zero, ub));
+	  optimizationAlgorithm_.setProblem(problemVI);
+	  multistart = MultiStart(optimizationAlgorithm_, startingPointsSupV);
+	  multistart.run();
+	  allMultiStarts.add(multistart);
+	  supV_[i] = std::exp(multistart.getResult().getOptimalValue()[0]);
+	  LOGDEBUG(OSS() << "supV_[" << i << "]=" << supV_[i] << " v*=" << multistart.getResult().getOptimalPoint());
+	} // if ub[i] > 0.0
+      if (lb[i] < 0.0)
+	{
+	  // find a feasible starting point in [lb, 0]
+	  if (!startingPointsInfV.getSize())
+	    {
+	      for (UnsignedInteger k = 0; k < candidateNumber_; ++ k)
+		{
+		  Point candidate(sequence.generate());
+		  for (UnsignedInteger j = 0; j < dimension; ++ j)
+		    candidate[j] = candidate[j] * lb[j];
+		  if (SpecFunc::IsNormal(logUnscaledPDF_(candidate)[0]))
+		    {
+		    startingPointsInfV.add(candidate);
+		    if (startingPointsInfV.getSize() == maximumMultistart_)
+		      break;
+		    }
+		} // for k
+	    } // (!startingPointsInfV.getSize())
+	  if (!startingPointsInfV.getSize())
+	    throw InternalException(HERE) << "Could not find a feasible starting point to initialize ratio of uniforms V inf";
+	  problemVI.setBounds(Interval(lb, zero));
+	  optimizationAlgorithm_.setProblem(problemVI);
+	  multistart = MultiStart(optimizationAlgorithm_, startingPointsInfV);
+	  multistart.run();
+	  allMultiStarts.add(multistart);
+	  infV_[i] = -std::exp(multistart.getResult().getOptimalValue()[0]);
+	  LOGDEBUG(OSS() << "infV_[" << i << "]=" << infV_[i] << " v*=" << multistart.getResult().getOptimalPoint());
+	} // if lb[i] < 0.0
+    } // for i
+  LOGDEBUG(OSS() << "supU multistart points=\n" << startingPointsSupU << "\ninfV multistart points=\n" << startingPointsInfV << "\nsupV multistart points=\n" << startingPointsSupV);
+<<<<<<< HEAD
```

Comment:
bad merge ?

Reply:
I tried to fix it a dozen times, but finally I had to switch commits to fix it definitely... Git, what a strange thing to me...